R/utils.R

```diff
@@ -1499,3 +1499,22 @@ levelIndex <- function(x, which = c("first", "middle", "last")) {
 .vdigest <- function(x, ...) {
     vapply(x, digest::digest, NA_character_, ..., USE.NAMES=FALSE)
 }
+
+#' Helper function to define indices for subsetting.
+#'
+#' @param x `character` being either names, rownames or column names.
+#'
+#' @param i `character`, `logical` or `numeric`.
+#'
+#' @author Johannes Rainer
+#'
+#' @noRd
+.to_index <- function(x, i, variable = "i") {
+    if (is.character(i))
+        i <- match(i, x)
+    if (is.logical(i))
+        i <- which(i)
+    if (any(is.na(i)) | any(i < 1) | any(i > length(x)))
```

Comment:
Please use `anyNA(i)` instead of `any(is.na(i))` and `||` instead of `|` (because we want to break ASOP, e.g. if any `i` is `NA` we don't want to test for `i < 1`):
```suggestion
    if (anyNA(i) || any(i < 1) || any(i > length(x)))
```